assets/opstools/AppBuilder/classes/platform/dataFields/ABFieldConnect.js

```diff
@@ -814,22 +814,67 @@ module.exports = class ABFieldConnect extends ABFieldConnectCore {
          options.filters = {};
       }
 
+      // if this field's options are filtered off another field's value we need
+      // to make sure the UX helps the user know what to do.
+      var placeholderReadOnly = null;
+      if (options.filterValue && options.filterKey) {
+         if (!$$(options.filterValue)) {
+            // this happens in the Interface Builder when only the single form UI is displayed
+            readOnly = true;
+            placeholderReadOnly =
+               "Must select item from 'PARENT ELEMENT' first.";
+         } else {
+            let val = this.getValue($$(options.filterValue));
+            if (!val) {
+               // if there isn't a value on the parent select element set this one to readonly and change placeholder text
+               readOnly = true;
+               let label = $$(options.filterValue);
+               placeholderReadOnly =
+                  "Must select item from '" + label.config.name + "' first.";
+            }
+         }
+      }
+
       // Render selectivity
       this.selectivityRender(
          domNode,
          {
             multiple: multiselect,
             data: selectedData,
-            placeholder: placeholder,
+            placeholder: placeholderReadOnly
+               ? placeholderReadOnly
+               : placeholder,
             readOnly: readOnly,
             editPage: options.editPage,
             ajax: {
                url: "It will call url in .getOptions function", // require
                minimumInputLength: 0,
                quietMillis: 250,
                fetch: (url, init, queryOptions) => {
+                  // if we are filtering based off another selectivity's value we
+                  // need to do it on fetch each time because the value can change
+                  var self = this;
+                  // copy the filters so we don't add to them every time there is a change
+                  var combineFilters = JSON.parse(
+                     JSON.stringify(options.filters)
+                  );
+                  // only add filters if we pass valid value and key
+                  if (options.filterValue && options.filterKey) {
+                     // get the current value of the parent select box
+                     let parentVal = self.getValue($$(options.filterValue));
```

Comment:
Are we able to use `this` directly instead of the `self` variable ?

Reply:
I can give it a try...I don't know why I am setting `var self = this;` on line 856. I will test it out and push changes if it works.